Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Document/ShaderManagementConsoleDocument.cpp

```diff
@@ -60,6 +60,33 @@ namespace ShaderManagementConsole
         ShaderManagementConsoleDocumentRequestBus::Handler::BusDisconnect();
     }
 
+    AZ::u32 ShaderManagementConsoleDocument::AddOneVariantRow()
+    {
+        auto shaderAssetResult =
+            AZ::RPI::AssetUtils::LoadAsset<AZ::RPI::ShaderAsset>(m_absolutePath,
+                                                                 m_shaderVariantListSourceData.m_shaderFilePath);
+        if (shaderAssetResult)
+        {
+            BeginEdit();
+            AZ::Data::Asset<AZ::RPI::ShaderAsset> shaderAsset = shaderAssetResult.GetValue();
+            AZStd::vector<AZ::RPI::ShaderOptionDescriptor> options = shaderAsset->GetShaderOptionGroupLayout()->GetShaderOptions();
+            AZ::RPI::ShaderVariantListSourceData::VariantInfo variantInfo;
+            for (auto& optionDescriptor : options)
+            {
+                variantInfo.m_options[optionDescriptor.GetName()] = ""; // all unset
+            }
```

Comment:
I'm not sure what other changes have been made with respect to shader options. Previously, I think the intention was to only add options to the map if the value was actually overridden. I don't remember exactly but I think that might have reduced the amount of serialized data. If the option was not in the map then it fell back to the default value. In that case the function could be reduced to resizing the shader variant vector and updating the stable IDs.

Reply:
I don't know if we should strive to keep the stableID compacts also. Because with the introduction of delete row, it creates gaps. Also, it's perfectly possible to make many rows that are similar in all points, that may mess up RHI internals if they work with equality/hashes. Maybe we need to run some verification to make the number of variants lean by ensuring uniquity. Also since verifying all against all risks to be quadratic, we can imagine to verify at save only. especially because during edition, it's ok to have temporary broken invariants.

Reply:
As discussed 2 weeks ago we can implement a "recompact" button.
But I'm also concerned about the builder, and the system that makes the economy of rebuilding all the variants by checking what stableID changed, if the stableID check doesn't incorporate the options's values, it will have lots of false negatives. We might have to extend the stableID with the whole key, even perhaps, replace it WITH the key to ensure natively unicity. But that could cause edition curiosities (like a difference between the view model and model)